plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1701,8 +1706,110 @@ public void testTruncateDecimalTransform()
+                        "DATE '2020-06-19'",
+                        "DATE '2020-07-19'",
+                        "DATE '2020-08-19'"),
+                ImmutableList.of(1, 2, 3, 4),
+                ImmutableList.of(
+                        "DATE '2020-09-19'",
+                        "DATE '2020-10-19'",
+                        "DATE '2020-11-19'"),
+                ImmutableList.of(5, 6, 7),
+                "CAST('2020-05-19' AS DATE)",
+                "CAST('2020-08-19' AS DATE)",
+                "CAST('2020-09-19' AS DATE)",
+                "CAST('2020-11-19' AS DATE)");
+
+        testBucketTransformForType(
+                "UUID",
+                ImmutableList.of(
+                        format("UUID '%s'", UUID.fromString("50000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("60000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("20000000-0000-0000-0000-000000000000"))),
+                ImmutableList.of(2, 3, 4),
+                ImmutableList.of(
+                        format("UUID '%s'", UUID.fromString("10000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("70000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("30000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("40000000-0000-0000-0000-000000000000"))),
+                ImmutableList.of(1, 5, 6, 7),
+                "CAST(null AS UUID)",
+                "CAST(null AS UUID)",
+                "CAST(null AS UUID)",
+                "CAST(null AS UUID)");
+
+        testBucketTransformForType(
+                "VARCHAR",
+                ImmutableList.of("'abcd'", "'mommy'", "'Grozny'"),
+                ImmutableList.of(1, 4, 7),
+                ImmutableList.of("'abxy'", "'ab598'", "'moscow'", "'Greece'"),
+                ImmutableList.of(2, 3, 5, 6),
+                "'Grozny'",
+                "'mommy'",
+                "'Greece'",
+                "'moscow'");
+
+        testBucketTransformForType(
+                "BIGINT",
+                ImmutableList.of("100000000", "200000002", "300000004", "700000007"),
+                ImmutableList.of(1, 4, 6, 7),
+                ImmutableList.of("400000001", "500000003", "600000005"),
+                ImmutableList.of(2, 3, 5),
+                "'100000000'",
+                "'700000007'",
+                "'400000001'",
+                "'600000005'");
+    }
+
+    private void testBucketTransformForType(
+            String type,
+            List<String> firstBucketKeys,
+            List<Integer> firstBucketValues,
+            List<String> secondBucketKeys,
+            List<Integer> secondBucketValues,
+            String firstBucketMinKey,
+            String firstBucketMaxKey,
+            String secondBucketMinKey,
+            String secondBucketMaxKey)
```

Comment:
This looks complex. What about having a method that works for single value at a time.

```
testBucketTransform(String type, String value, int bucketCount, int expectedBucket, String valueInOtherBucket)
```

or  -- if we want to test min/max-es too:
```
testBucketTransform(String type, String value, int bucketCount, int expectedBucket, String greaterValueInSameBucket, String valueInOtherBucket)
```

Reply:
I wanted to stay as close to the original test as possible but it maybe it makes sense to make it simpler